pytest.ini

```diff
@@ -0,0 +1,3 @@
+[pytest]
+addopts = -v -ra --cov
+python_paths = src/
```

Comment:
Is this line necessary?  It isn't present in the skeleton.

Reply:
@dav3r I couldn't find an easy way to not include that. When I didn't have it, I was getting a `ModuleNotFound` error on these lines when running the pytest.

https://github.com/cisagov/domain-manager-cli/blob/f194b492d0641e6bf3ea72d8e55ccdcd937fb742/tests/test_dmcli.py#L12-L13

In the past, the easiest work around I've done is just using `pytest-pythonpath`. If you have another idea to make this work, or know of something I may be doing wrong, I'm all for it.

Reply:
Odd.  Once I set up the local Python virtual environment it works for me out of the box:
```
$ pyenv virtualenv 3.9.5 domain-manager-cli
$ pyenv local domain-manager-cli
$ pip install --upgrade pip setuptools wheel && pip install --upgrade -r requirements-dev.txt && pre-commit install
$ pytest
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.9.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- /home/jeremy_frasier/.pyenv/versions/3.9.5/envs/domain-manager-cli/bin/python3.9
cachedir: .pytest_cache
rootdir: /home/jeremy_frasier/cisagov/domain-manager-cli, configfile: pytest.ini
plugins: pythonpath-0.7.3, cov-2.11.1
collected 1 item                                                                                                                                                                                                                             

tests/test_dmcli.py::test_stdout_version PASSED                                                                                                                                                                                        [100%]

----------- coverage: platform linux, python 3.9.5-final-0 -----------
Name                                  Stmts   Miss Branch BrPart  Cover   Missing
---------------------------------------------------------------------------------
src/dmcli/__init__.py                     1      0      0      0   100%
src/dmcli/__main__.py                     2      2      0      0     0%   4-6
src/dmcli/_version.py                     1      0      0      0   100%
src/dmcli/groups/__init__.py              0      0      0      0   100%
src/dmcli/groups/application.py          10      3      2      0    58%   13, 19-20
src/dmcli/groups/category.py             10      3      2      0    58%   13, 19-20
src/dmcli/groups/configure.py            17      6      0      0    65%   17-26
src/dmcli/groups/domain.py              110     67     30      0    31%   27, 34-39, 45-46, 54-55, 64-83, 90-92, 102-105, 112-115, 122-124, 131-133, 140-147, 154-162, 174-190
src/dmcli/groups/external.py             21      9      8      0    41%   14, 21-23, 30-36
src/dmcli/groups/record.py               73     50     24      0    24%   13, 20-39, 77-87, 95-98, 106-116, 121-147
src/dmcli/groups/template.py             16      6      4      0    50%   13, 19-20, 27-29
src/dmcli/main.py                        29      4      2      0    81%   32-35
src/dmcli/utils/__init__.py               0      0      0      0   100%
src/dmcli/utils/api.py                   43     34      2      0    20%   13-14, 21-28, 33-40, 45-52, 57-64
src/dmcli/utils/application.py            6      4      2      0    25%   8-11
src/dmcli/utils/categories.py             3      1      0      0    67%   8
src/dmcli/utils/domain.py                35     20      2      0    41%   11-14, 19, 24, 31-37, 42, 47, 55-58, 66, 71, 76, 81, 88, 93
src/dmcli/utils/external.py               5      2      0      0    60%   8, 13
src/dmcli/utils/message_handling.py      13      6      0      0    54%   8, 13, 18, 23, 28, 33
src/dmcli/utils/settings.py               9      0      2      1    91%   13->19
src/dmcli/utils/templates.py             10      5      0      0    50%   11, 16, 21-23
---------------------------------------------------------------------------------
TOTAL                                   414    222     80      1    39%


============================================================================================================= 1 passed in 0.43s ==============================================================================================================
```

Reply:
I think the issue is in these lines:
https://github.com/cisagov/domain-manager-cli/blob/b3ee7cf68f5192c0a71f22eec252826b1debefc6/setup.py#L84-L86

Notice how they are different from the [corresponding lines in the skeleton](https://github.com/cisagov/skeleton-python-library/blob/b85e1b85ce7816b8961c0afb2fb9b843ba6e96a5/setup.py#L85-L88).  I think removing the `package_data` line is fine, but the others are affecting the path where python is looking for your modules.

Keep in mind that making this change means you will have to change how you reference all of your modules within `dmcli` (e.g. `from groups.configure import configure` --> `from dmcli.groups.configure import configure`).

Reply:
Ah, I get the failure you mentioned if I remove the `python_paths = src/` line.  I wonder why [cisagov/skeleton-python-library](https://github.com/cisagov/skeleton-python-library) does not need that line but this project does?  Any thoughts @mcdonnnj?

Reply:
@jsf9k - Take a look at [my comment above](https://github.com/cisagov/domain-manager-cli/pull/75#discussion_r631312356) that you may have missed.

Reply:
@dav3r Did you remove the `python_paths`? This is the result when I remove it and run the same commands you did above.

```
============================================================================================================================ test session starts =============================================================================================================================
platform linux -- Python 3.9.4, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- /home/wilszs/.pyenv/versions/3.9.4/envs/domain-manager-cli/bin/python3.9
cachedir: .pytest_cache
rootdir: /home/wilszs/projects/domain-manager-cli, configfile: pytest.ini
plugins: cov-2.11.1
collected 0 items / 1 error
Coverage.py warning: No data was collected. (no-data-collected)

=================================================================================================================================== ERRORS ===================================================================================================================================
____________________________________________________________________________________________________________________ ERROR collecting tests/test_dmcli.py ____________________________________________________________________________________________________________________
ImportError while importing test module '/home/wilszs/projects/domain-manager-cli/tests/test_dmcli.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../.pyenv/versions/3.9.4/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_dmcli.py:12: in <module>
    import dmcli
E   ModuleNotFoundError: No module named 'dmcli'

----------- coverage: platform linux, python 3.9.4-final-0 -----------
```

Reply:
Yeah, I think @dav3r identified the problem.

Reply:
@dav3r Just saw your comments as well. I was having the same issue when those lines were what they are originally in (here)[https://github.com/cisagov/skeleton-python-library/blob/b85e1b85ce7816b8961c0afb2fb9b843ba6e96a5/setup.py#L85-L88].

Reply:
@dav3r Actually, don't get the same error. This is what I get instead.

```
=================================================================================================================================== ERRORS ===================================================================================================================================
____________________________________________________________________________________________________________________ ERROR collecting tests/test_dmcli.py ____________________________________________________________________________________________________________________
ImportError while importing test module '/home/wilszs/projects/domain-manager-cli/tests/test_dmcli.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../.pyenv/versions/3.9.4/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_dmcli.py:13: in <module>
    from dmcli import main
src/dmcli/main.py:3: in <module>
    from _version import __version__
E   ModuleNotFoundError: No module named '_version'
```

And here is what the setup.py lines are with that.

```
keywords="domain-manager",
packages=find_packages(where="src/"),
package_dir={"": "src/"},
py_modules=[splitext(basename(path))[0] for path in glob("src/*.py")],
```

Reply:
Change `from _version import __version__` to `from ._version import __version__`

Reply:
@dav3r @jsf9k Yeah, I just realized that. I have to change it `from _version import __version__` to from `._version import __version__`

But I also have to change all the other imports to - `from groups.category import category` to `from dmcli.groups.category import category`

That should make everything work properly.

Reply:
@dav3r @jsf9k Made those fixes. Let me know if everything looks good in https://github.com/cisagov/domain-manager-cli/pull/75/commits/7d299d9da367c686b86d6bf8164dda11711e5e74 after the checks pass.

Reply:
Looks good to me.  Nice work!